### ISSUE TITLE
How to consolidate maven coordiantes for past CQs?

### ISSUE THREAD

**Issue by boaks** [eclipse/dash-licenses#32]
I really like this tool! 

using it for Eclipse/Californium, I got aware, that some CQs are not found:

> maven/mavencentral/com.google.code.gson/gson/2.8.2, Apache-2.0, approved, CQ14005
> maven/mavencentral/com.upokecenter/cbor/4.0.0, unknown, restricted, none

I already created a CQ for that a year ago, see [CQ21057](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21057). I used the summary "com.upokecenter.cbor-4.0.0".
I don't know, how this tool locates CQs from the `IContentId`, but if there is a possibility to consolidate that mapping information, let me know (e.g. a APi to add the maven-coordinates to past CQs).

**Comment by waynebeaton** [comment]
TL;DR: I've added a mapping that should fix your immediate problem.

Long winded explanation:

The source of the problem is that our IPZilla system doesn't do a good job of matching the technical name to the CQ. 

We made a change to the language in the CQ creation page that encourages you to use the Maven GAV (or NPM/ClearlyDefined ID) as the CQ name. In cases where an ID is used, the mapping should happen automatically. So... for example, if you'd set the Summary of the CQ to "com.upokecenter:cbor:4.0.0" on CQ 21057, the backend that the license tool uses would be able to automatically make the mapping. At some point, I'm going to have the IP Team go through all of our historical records and fix the Summaries (where possible).

One of the challenges that we have, especially with historical data, is that some CQs actually represent multiple JAR files, so one single mapping isn't sufficient. This is a consideration that we're keeping in mind as we contemplate the replacement for IPZilla.

In the meantime have a couple of internal hacks that we use to track these mappings, but they require that we become aware of the missing mapping and add it. I've added this one to that mapping. The script that pulls all of these mappings together runs a few times a day, so that particular example should drop off your list if you re-run the tool sometime later today or tomorrow.

I'd like to leave this issue open for the time being. I expect that I'll end up opening a related issue against IPZilla, but want to think about it a bit first.

**Comment by boaks** [comment]
Thanks for adding cbor 4.0.0!  

For Californium itself, I get now only one "none" left:

```
maven/mavencentral/com.github.peteroupc/numbers/1.4.3, unknown, restricted, none
```

**Comment by waynebeaton** [comment]
Added. The script that pulls data together for the back-end runs twice a day. That mapping should be recognized by the back-end later today.

**Comment by waynebeaton** [comment]
FYI, I noticed that https://github.com/eclipse/californium.git has a third-party submodule and didn't see an obvious CQ for it. I believe that it's a Hugo theme. Can you tell me what it's used for?

**Comment by boaks** [comment]
@sophokles73 , maybe you know more about that?

AFAIK, it's used to generate the projects web-site [Eclipse/Californium](https://www.eclipse.org/californium/).

**Comment by sophokles73** [comment]
Exactly, it is used as part of the nightly build on Jenkins to generate the project website from markdown source code that is part of the Californium GitHub repo.

**Comment by thjaeckle** [comment]
@waynebeaton is this issue meant to report more of past CQs which are not yet mapped to maven coordinates?
Would be great to have a process to do so (as IPZilla request or as GitHub issue in this repo, maybe with a template).

For the time being I have 
`org.jetbrains:annotations:16.0.3` 
`maven/mavencentral/org.jetbrains/annotations/16.0.3`
not being resolved as CQ https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20715

**Comment by waynebeaton** [comment]
> @waynebeaton is this issue meant to report more of past CQs which are not yet mapped to maven coordinates?

No. Not really.

> Would be great to have a process to do so (as IPZilla request or as GitHub issue in this repo, maybe with a template).

I normally resolve these mapping issues when I do IP Log reviews.

One thing that you can do is to use either Maven GAV or the ClearlyDefined coordinates as the summary when you create the CQ (we've tweaked the UI to recommend this).

Clearly though, we need to sort out a consistent means of interfacing with the IP Team to update this information. I've been working with them, but don't have a firm plan yet. 

> 
> For the time being I have
> `org.jetbrains:annotations:16.0.3`
> `maven/mavencentral/org.jetbrains/annotations/16.0.3`
> not being resolved as CQ https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20715

I've updated the CQ summary. This should resolve after the aggregator script runs (it runs at 3am and 2pm ET daily; I may bump up the frequency of that).

**Comment by waynebeaton** [comment]
I think that we're done here. Reopen if I've missed something.
